extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/CaseInsensitiveStringMatchPushdownHandler.java

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.parquet.table.location;
+
+import io.deephaven.engine.table.impl.select.MatchFilter;
+import org.apache.commons.lang3.mutable.MutableObject;
+import org.apache.parquet.column.statistics.Statistics;
+import org.jetbrains.annotations.NotNull;
+
+final class CaseInsensitiveStringMatchPushdownHandler {
+
+    static boolean maybeOverlaps(
+            @NotNull final MatchFilter matchFilter,
+            @NotNull final Statistics<?> statistics) {
+        final Object[] values = matchFilter.getValues();
+        final boolean inverseMatch = matchFilter.getInvertMatch();
+        if (values == null || values.length == 0) {
+            // No values to check against, so we consider it as a maybe overlap.
+            return true;
+        }
+        final Comparable<?>[] comparableValues = new Comparable<?>[values.length];
+        for (int i = 0; i < values.length; i++) {
+            if (!(values[i] instanceof String)) {
+                // Skip pushdown-based filtering for nulls to err on the safer side instead of adding more complex
+                // handling logic.
+                // TODO (DH-19666): Improve handling of nulls
+                return true;
+            }
+            // Convert to lower case for case-insensitive comparison
+            comparableValues[i] = ((String) values[i]).toLowerCase();
```

Comment:
I don't think we should be converting values to accomplish this. It's not obvious to me if it produces the same results as opposed to `java.lang.String#compareToIgnoreCase` (which I assume is consistent with `java.lang.String#equalsIgnoreCase`, which `io.deephaven.engine.table.impl.chunkfilter.StringChunkMatchFilterFactory#makeCaseInsensitiveFilter` uses); so we should use `compareToIgnoreCase`.

Note: I think there is a bug in `io.deephaven.base.string.cache.CharSequenceUtils#caseInsensitiveHashCode`. I will follow-up with a separate issue.